### PR TITLE
Add TSDB support to grafana-loki chart

### PR DESCRIPTION
**Pull request by pahanik**
TSDB is the recommended Loki index since version 2.8. This configuration is based on an example configuration from https://grafana.com/docs/loki/latest/operations/storage/tsdb/

### Applicable issues

- fixes #

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/grafana-loki/values.yaml [hunk 1]
```diff
@@ -164,6 +172,27 @@ loki:
           addresses: dns+{{ include "grafana-loki.memcached-index-queries.host" . }}
           service: http
         {{- end }}
+      # New tsdb-shipper configuration
```
Comment (andresbono) [inline bitnami/grafana-loki/values.yaml:175]: Please, remove the comments in this configuration file, at least those not strictly necessary. They make sense in [TSDB: Example Configuration](https://grafana.com/docs/loki/latest/operations/storage/tsdb/#example-configuration) for illustrative purposes but I think they are not necessary here.

**Review by andresbono — COMMENTED** [review]
Hi @pahanik, thank you very much for your contribution! Can you think of any drawbacks or backwards compatibility issues for existing users of this chart?

File: bitnami/grafana-loki/values.yaml (hunk 1 above)
Comment (pahanik) [reply bitnami/grafana-loki/values.yaml]: Done

Comment (andresbono) [reply bitnami/grafana-loki/values.yaml]: Thank you for addressing this.

**Review by andresbono — DISMISSED** [review]
LGTM now, thank you! Can you think of any drawbacks or backwards compatibility issues for existing users of this chart?

**Comment by pahanik** [comment]
> LGTM now, thank you! Can you think of any drawbacks or backwards compatibility issues for existing users of this chart?

NP. I haven't enabled caching like index_queries_cache_config and cache_ttl like it is done in storage_config.boltdb_shipper, cause I haven't tested such configurations yet. For me, tsdb works better with Grafana because logs are shown for long time frames in contrast to boltdb where only the last 2 hours are shown.

**Comment by andresbono** [comment]
> I haven't enabled caching like index_queries_cache_config and cache_ttl like it is done in storage_config.boltdb_shipper, cause I haven't tested such configurations yet. According to https://grafana.com/docs/loki/latest/operations/storage/tsdb/ "Index Caching not required".

Please note that `index_queries_cache_config` is a global configuration to both shippers. I wonder if we should plan a future change in the chart deactivating the index cache config to false:

```diff
 ## @section Memcached Sub-chart Parameters (Index-Queries)
 ## Memcached sub-chart (Index-Queries)
 ##
 memcachedindexqueries:
   ## @param memcachedindexqueries.enabled Deploy memcached sub-chart
   ##
-  enabled: true
+  enabled: false
```

> Maybe, we could set something like e.g. '- from: "2024-02-01"' in schema_config.configs part related to TSDB and add to the output of "helm install" command something like: "WARNING: grafana-loki will be switched to TSDB on February, 1. Please see https://grafana.com/docs/loki/latest/operations/storage/tsdb/"

I think it's a good idea 👍 If you agree on planning that other change deactivating the index cache config, we can remove the warning message in that future release.

File: bitnami/grafana-loki/values.yaml [hunk 2]
```diff
@@ -141,6 +141,13 @@ loki:
         index:
           prefix: index_
           period: 24h
+      - from: "2024-01-20"
```
[the remark this answers is not in the document]
Comment (pahanik) [reply bitnami/grafana-loki/values.yaml:144]: Which date do you suggest? February, 1 is already outdated.

**Comment by pahanik** [comment]
> > I haven't enabled caching like index_queries_cache_config and cache_ttl like it is done in storage_config.boltdb_shipper, cause I haven't tested such configurations yet. According to https://grafana.com/docs/loki/latest/operations/storage/tsdb/ "Index Caching not required".
> 
> Please note that `index_queries_cache_config` is a global configuration to both shippers. I wonder if we should plan a future change in the chart deactivating the index cache config by default:
> 
> ```diff
>  ## @section Memcached Sub-chart Parameters (Index-Queries)
>  ## Memcached sub-chart (Index-Queries)
>  ##
>  memcachedindexqueries:
>    ## @param memcachedindexqueries.enabled Deploy memcached sub-chart
>    ##
> -  enabled: true
> +  enabled: false
> ```
> 
> > Maybe, we could set something like e.g. '- from: "2024-02-01"' in schema_config.configs part related to TSDB and add to the output of "helm install" command something like: "WARNING: grafana-loki will be switched to TSDB on February, 1. Please see https://grafana.com/docs/loki/latest/operations/storage/tsdb/"
> 
> I think it's a good idea 👍 If you agree on planning that other change deactivating the index cache config, we can remove the warning message in that future release.

So, what do you suggest? Am I correct to assume that first we commit tsdb related config with some future date and then we commit memcachedindexqueries.enabled=false on that date?

File: bitnami/grafana-loki/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (andresbono) [reply bitnami/grafana-loki/values.yaml:144]: Maybe 2024-03-12 (Tuesday)?

**Comment by andresbono** [comment]
> According to https://grafana.com/docs/loki/latest/operations/storage/tsdb/#index-caching-not-required: "If you are already using Loki with other index types, it is recommended to keep the index caching until all of your existing data falls out of [retention](https://grafana.com/docs/loki/latest/operations/storage/retention/) or your configured max_query_lookback under [limits_config](https://grafana.com/docs/loki/latest/configure/#limits_config). After that, we suggest running without an index cache (it isn’t used in TSDB)." Neither retention nor max_query_lookback is configured in https://github.com/bitnami/charts/blob/main/bitnami/grafana-loki/values.yaml. Maybe, let's also set retention for a couple of weeks and then disable caching.

That makes sense to me 👍

File: bitnami/grafana-loki/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pahanik) [reply bitnami/grafana-loki/values.yaml]: Done.

**Comment by carrodher** [comment]
Sorry for the inconvenience, there was a new minor version of this chart that requires a rebase from `main` and a new bump of the version

**Comment by andresbono** [comment]
This is now merged 🎉 Thank you for your contribution. Would you like to open the follow-up PR planned for 2024-03-26?

**Pull request merged**